test/files/run/t11272.scala

```diff
@@ -0,0 +1,12 @@
+
+object Test {
+  def main(args: Array[String]): Unit = {
+    test()
+  }
+  def test() = {
+    val iter = Iterator(128*1024*1024, 128*1024*1024).flatMap(new Array[Byte](_))
```

Comment:
Does this test actually do anything? Do we have any guarantees that it is running with sufficiently little memory that incorrect behaviour would actually cause the test to fail (presumably via OOM)?

Reply:
There is a .javaopts file.

Reply:
oops, I missed that

Reply:
I didn't merge this forward. It was mostly to convince myself, since the unit test was tricky to get right (until I realized that it turns on how array iterator works).